django/db/migrations/state.py

```diff
@@ -304,12 +304,23 @@ def __init__(self, app_label, name, fields, options=None, bases=None, managers=N
         # Sanity-check that fields is NOT a dict. It must be ordered.
         if isinstance(self.fields, dict):
             raise ValueError("ModelState.fields cannot be a dict - it must be a list of 2-tuples.")
-        # Sanity-check that fields are NOT already bound to a model.
         for name, field in fields:
+            # Sanity-check that fields are NOT already bound to a model.
             if hasattr(field, 'model'):
                 raise ValueError(
                     'ModelState.fields cannot be bound to a model - "%s" is.' % name
                 )
+            # Sanity-check that relation fields are NOT referring to a model class.
+            if field.remote_field and hasattr(field.remote_field.model, '_meta'):
```

Comment:
Any reason why we can't go with `field.is_relation` and `field.related_model` here?